AutoCollection/PreAggregatedMetrics.ts

```diff
@@ -0,0 +1,301 @@
+import TelemetryClient = require("../Library/TelemetryClient");
+import Constants = require("../Declarations/Constants");
+
+import { AggregatedMetric } from "../Declarations/Metrics/AggregatedMetric";
+import { AggregatedMetricCounter } from "../Declarations/Metrics/AggregatedMetricCounters";
+import {
+    MetricBaseDimensions,
+    MetricDependencyDimensions,
+    MetricExceptionDimensions,
+    MetricRequestDimensions,
+    MetricTraceDimensions,
+    PreaggregatedMetricPropertyNames,
+    MetricDimensionTypeKeys
+} from "../Declarations/Metrics/AggregatedMetricDimensions";
+import * as Contracts from "../Declarations/Contracts";
+
+
+class AutoCollectPreAggregatedMetrics {
+
+    public static INSTANCE: AutoCollectPreAggregatedMetrics;
+    private _collectionInterval: number;
+    private static _client: TelemetryClient;
+    private _handle: NodeJS.Timer;
+    private _isEnabled: boolean;
+    private _isInitialized: boolean;
+    private static _dependencyCountersCollection: Array<AggregatedMetricCounter>;
```

Comment:
I'm wondering why you need to a keep a collection of counters? We only need one counter per request type.

Reply:
AggregatedMetricCounter here is a class that have all necessary counters plus the dimensions, we need a collection to send all data by dimensions